File: api/src/main/java/com/api/app/emum/DLV001.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum DLV001 implements CommonCode {
    SHIPMENT("001", "출하", 1, "", ""),
    COLLECTION("002", "회수", 2, "", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static DLV001 findByCode(String code) {
        return CommonCodeUtil.findByCode(DLV001.class, code);
    }
}

File: api/common-lib/common-core/src/main/kotlin/com/api/app/emum/DLV001.kt
package com.api.app.emum

/** 배송유형코드: deliveryTypeCode */
enum class DLV001(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    DELIVERY("001", "일반배송", 1, "", ""),
    COLLECTION("002", "반품배송", 2, "", ""),
    ;

    companion object {
        fun findByCode(code: String): DLV001? = CommonCodeUtil.findByCode(DLV001::class.java, code)
    }
}
